src/App.tsx

```diff
@@ -1,34 +1,117 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { filteringTodo } from './services/filteringTodos';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[] | null>(null);
+  const [query, setQuery] = useState('');
+  const [selectFiltering, setselectFiltering] = useState('');
+  const [filteredTodos, setFilteredTodos] = useState<Todo[] | null>(todoList);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isActive, setIsActive] = useState(false);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoading(true);
+
+      try {
+        const todos = await getTodos();
+
+        setTodoList(todos);
+        setFilteredTodos(todos);
+      } catch (err) {
+        setTodoList(null);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchTodos();
+  }, []);
+
+  const handleQueryChange = (value: string) => {
+    setQuery(value.trim().toLowerCase());
+  };
+
+  const handleSelectChange = (value: string) => {
+    setselectFiltering(value);
```

Comment:
The function name `setselectFiltering` does not follow the camelCase naming convention. Consider renaming it to `setSelectFiltering` to adhere to standard naming conventions.